### ISSUE TITLE
Different behavior of script invocation from Job task on Windows and Linux

### ISSUE THREAD

**Issue by 237dmitry** [PowerShell/PowerShell#20935]
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the [latest released version](https://github.com/PowerShell/PowerShell/releases)
- [X] Search the existing issues.
- [X] Refer to the [FAQ](https://github.com/PowerShell/PowerShell/blob/master/docs/FAQ.md).
- [X] Refer to [Differences between Windows PowerShell 5.1 and PowerShell](https://learn.microsoft.com/powershell/scripting/whats-new/differences-from-windows-powershell).

### Steps to reproduce

I use script for listening music in console background. The script is long, this is a main part of it:

```powershell
$PlayJob = Start-Job { play --replay-gain track $using:track } -Name 'sox'

Register-ObjectEvent -InputObject $PlayJob -EventName StateChanged -Action {

    Unregister-Event $EventSubscriber.SourceIdentifier
    Remove-Job $EventSubscriber.SourceIdentifier
    Remove-Job -Id $EventSubscriber.SourceObject.Id
    Remove-Variable -Name track, fstate, info -Scope Global

    if ($Global:fulllist)
    {
        if ($Global:playlist.Count -gt 0) { fplay }              # fplay is the script name
        else { Remove-Variable fulllist -Scope Global }
    }

} | Out-Null
```

In Windows everything works as expected, the script runs itself. But in Linux I need to press any arrow key on the keyboard and then it will start playing. I don't understand why this is happening. I tested in several terminal emulators and in the Linux tty console.

### Expected behavior

```console
Calling the script again should work the same on both OSes.
```

### Actual behavior

```console
- on Windows works ideally
- on Linux I have to press any of Arrow keys
```

### Error details

```console
No errors
```

### Environment data

```powershell
Name                           Value
----                           -----
PSVersion                      7.4.0
PSEdition                      Core
GitCommitId                    7.4.0
OS                             ArchLinux
Platform                       Unix
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

**Comment by SeeminglyScience** [comment]
This is due to a bug (sort of) in PSReadLine. `ReadKey` is ran in the background, and on non-Windows platforms that creates a BCL internal lock on stdin. When PSRL tries to process events in its input loop, it tries to check the current cursor position which also tries to acquire the same lock, creating a deadlock until a key is pressed.

@daxian-dbw I remember talking to Jason about this back when I was adding changes to get it working in PSES. Do you know if this is already being tracked somewhere? If not I'm happy to open a detailed issue (though I don't if it's feasible to fix without just skipping the cursor position check, or adding the `ReadKey` impl PSES injects).

**Comment by 237dmitry** [comment]
I tried to emulate keystrokes using `xdotool`, but it only returns `^[[A^[[B]` to the host and no emulation. I can't do this with `System.Windows.Forms` as it is not available on Linux.

**Comment by rhubarb-geek-nz** [comment]
Is PSRL being used by the asynchronous job? I would have expected that PSRL is used by the console for interactive input from the user but jobs should be detached from the console much like when running nohup in the background.

**Comment by SeeminglyScience** [comment]
> Is PSRL being used by the asynchronous job?

No, it's being used by the original process/thread. The job will work fine, it's the handling of the event that won't
